src/ssm-mem.c

```diff
@@ -4,13 +4,43 @@
  *  @author John Hui (j-hui)
  */
 #include <ssm-internal.h>
-
+#include <allocationDispatcher.h>
+ALLOCATION_DISPATCHER *ad = NULL;
+size_t allSizes[1000];
+int numSizes = 0;
 struct ssm_mm *ssm_mem_alloc(size_t size) {
-  return malloc(size); // TODO: (dan) use our own allocator
+  if(ad==NULL){
```

Comment:
Adding this check every single time is unnecessarily costly. Initialization should be performed in a separate function. Add an `ssm_mem_init()` function for that.